src/tests/unittests/configuration_helper/test_instrument_adapter_factory.py

```diff
@@ -36,33 +40,11 @@ def test_raise_value_error(self):
         with self.assertRaises(ValueError):
             InstrumentAdapterFactory.get_instrument_adapter('SomeAdapter', 'dev42')
 
-    def test_import_error_as_value_error(self):
-        import builtins
-        import sys
-        import importlib
-        importlib.invalidate_caches()
+    def test_external_adapters(self):
+        self.assertRaises(ValueError, InstrumentAdapterFactory.get_instrument_adapter, 'DummyInstrumentAdapter', '')
 
-        current_modules = sys.modules.copy()
-        original_import = builtins.__import__
-
-        def fake_import(name, *args, **kwargs):
-            if name.startswith('qtt'):
-                raise ImportError()
-            else:
-                return original_import(name, *args, **kwargs)
-
-        builtins.__import__ = fake_import
-
-        modules = {'TimeStampInstrumentAdapter': 'qilib.configuration_helper.adapters.time_stamp_instrument_adapter'}
-        for path in modules.values():
-            del sys.modules[path]
-
-        reload(adapters)
-
-        for adapter in modules.keys():
-            error_msg = f'Failed to load {adapter}'
-            get_adapter = InstrumentAdapterFactory.get_instrument_adapter
-            self.assertRaisesRegex(ValueError, error_msg, get_adapter, adapter, 'address')
+        import sys
+        InstrumentAdapterFactory.add_instrument_adapters(sys.modules[__name__])
 
-        builtins.__import__ = original_import
-        sys.modules = current_modules.copy()
+        adapter = InstrumentAdapterFactory.get_instrument_adapter('DummyInstrumentAdapter', '')
+        self.assertTrue(issubclass(type(adapter), InstrumentAdapter))
```

Comment:
something like: `self.assertIsInstance(adapter, DummyInstrumentAdapter)` is cleaner and more explicit

Reply:
That was indeed what was meant!